### PR TITLE
Fix load_balancer_pool acceptance tests

### PR DESCRIPTION
**Pull request by mgirouard**
```
mgirouard@sixteen:~/src/terraform-provider-cloudflare$ TF_ACC=1 go test ./internal/services/load_balancer_pool -run "^TestAcc" -v -count=1
=== RUN   TestAccCloudflareLoadBalancerPools
--- PASS: TestAccCloudflareLoadBalancerPools (3.84s)
=== RUN   TestAccCloudflareLoadBalancerPool_Basic
=== PAUSE TestAccCloudflareLoadBalancerPool_Basic
=== RUN   TestAccCloudflareLoadBalancerPool_OriginSteeringLeastOutstandingRequests
=== PAUSE TestAccCloudflareLoadBalancerPool_OriginSteeringLeastOutstandingRequests
=== RUN   TestAccCloudflareLoadBalancerPool_OriginSteeringLeastConnections
=== PAUSE TestAccCloudflareLoadBalancerPool_OriginSteeringLeastConnections
=== RUN   TestAccCloudflareLoadBalancerPool_VirtualNetworkID
=== PAUSE TestAccCloudflareLoadBalancerPool_VirtualNetworkID
=== RUN   TestAccCloudflareLoadBalancerPool_FullySpecified
=== PAUSE TestAccCloudflareLoadBalancerPool_FullySpecified
=== RUN   TestAccCloudflareLoadBalancerPool_CreateAfterManualDestroy
=== PAUSE TestAccCloudflareLoadBalancerPool_CreateAfterManualDestroy
=== CONT  TestAccCloudflareLoadBalancerPool_Basic
=== CONT  TestAccCloudflareLoadBalancerPool_VirtualNetworkID
=== CONT  TestAccCloudflareLoadBalancerPool_OriginSteeringLeastConnections
=== CONT  TestAccCloudflareLoadBalancerPool_CreateAfterManualDestroy
=== CONT  TestAccCloudflareLoadBalancerPool_OriginSteeringLeastOutstandingRequests
=== CONT  TestAccCloudflareLoadBalancerPool_FullySpecified
--- PASS: TestAccCloudflareLoadBalancerPool_OriginSteeringLeastOutstandingRequests (2.57s)
--- PASS: TestAccCloudflareLoadBalancerPool_OriginSteeringLeastConnections (2.67s)
--- PASS: TestAccCloudflareLoadBalancerPool_FullySpecified (2.71s)
--- PASS: TestAccCloudflareLoadBalancerPool_Basic (2.79s)
--- PASS: TestAccCloudflareLoadBalancerPool_CreateAfterManualDestroy (3.80s)
--- PASS: TestAccCloudflareLoadBalancerPool_VirtualNetworkID (9.37s)
PASS
ok      github.com/cloudflare/terraform-provider-cloudflare/internal/services/load_balancer_pool        13.227s
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -8,7 +8,7 @@ require (
 	github.com/aws/aws-sdk-go-v2/credentials v1.17.27
 	github.com/aws/aws-sdk-go-v2/service/s3 v1.58.2
 	github.com/cloudflare/cloudflare-go v0.100.0
-	github.com/cloudflare/cloudflare-go/v2 v2.4.0
```
Comment (jacobbednarz) [inline go.mod:11]: let's uncommit this to avoid breaking the reference in the config.

File: internal/services/load_balancer_pool/model.go [hunk 2]
```diff
@@ -28,9 +28,9 @@ type LoadBalancerPoolModel struct {
 	LoadShedding       customfield.NestedObject[LoadBalancerPoolLoadSheddingModel]       `tfsdk:"load_shedding" json:"load_shedding,computed_optional"`
 	NotificationFilter customfield.NestedObject[LoadBalancerPoolNotificationFilterModel] `tfsdk:"notification_filter" json:"notification_filter,computed_optional"`
 	OriginSteering     customfield.NestedObject[LoadBalancerPoolOriginSteeringModel]     `tfsdk:"origin_steering" json:"origin_steering,computed_optional"`
-	CreatedOn          timetypes.RFC3339                                                 `tfsdk:"created_on" json:"created_on,computed" format:"date-time"`
+	CreatedOn          types.String                                                      `tfsdk:"created_on" json:"created_on,computed"`
```
Comment (jacobbednarz) [inline internal/services/load_balancer_pool/model.go:31]: are we fixing this in the config? or running with a custom code option here?

File: go.mod (hunk 1 above)
Comment (mgirouard) [reply go.mod]: Reverted in d82b934

File: internal/services/load_balancer_pool/model.go (hunk 2 above)
Comment (mgirouard) [reply internal/services/load_balancer_pool/model.go:31]: Updated in the config and it looks like it codegen'd correctly.

**Pull request merged**